### PR TITLE
update create-client doc for react

### PR DESCRIPTION
**Pull request by Hidoyatmz**
I tried to create a client following the [current documentation](https://api-platform.com/docs/create-client/react/) and using `npm init react-app -- --template typescript my-app` but ran into a couple of problems.

I could not find the file `src/index.tsx` as in the doc but only a `src/index.js`
I first tried to convert this file into a `.tsx` file but when launching the app with `npm run` (react-scripts start) an error occured saying `Could not find a required file (index.js)`

As we created the app with the parameter `--template typescript` I think that these modifications should not be necessary and something wrong is going on
After doing the same steps with vite bundle which use the latest recommended libraries, I found out everything was going well and I had no problems setting up the application by following the current documentation

### CODE REVIEW

**Comment by dunglas** [comment]
https://github.com/api-platform/docs/pull/1911/files#diff-40f90ff67f40f9806065e8ef506ed90ba282c3ecf369b7078816715b2c3bb3a9R12 should also be updated because we don't use CRA anymore.

**Comment by Hidoyatmz** [comment]
> https://github.com/api-platform/docs/pull/1911/files#diff-40f90ff67f40f9806065e8ef506ed90ba282c3ecf369b7078816715b2c3bb3a9R12 should also be updated because we don't use CRA anymore.

True, removed it

**Review by dunglas — CHANGES_REQUESTED** [review]

File: create-client/react.md [hunk 1]
```diff
@@ -74,9 +72,9 @@ root.render(
   <React.StrictMode>
     <Router>
       <Routes>
-        <Route path="/" component={Welcome} strict={true} exact={true}/>
+        <Route path="/" element={<App/>}/>
         {/* Add your routes here */}
-        <Route render={() => <h1>Not Found</h1>} />
+        <Route path='*' element={<NotFound/>} />
```
[the remark this answers is not in the document]
Comment (Hidoyatmz) [reply create-client/react.md:77]: The NotFound element is already defined in the given example from the documentation (https://github.com/api-platform/docs/blob/3.2/create-client/react.md?plain=1#L66)

It was not used, that's why I changed this line to use it

**Review by dunglas — APPROVED** [review]

**Comment by dunglas** [comment]
Thanks!!

**Pull request merged**